internal/batches/service/service.go

```diff
@@ -185,96 +184,75 @@ func (svc *Service) EnsureImage(ctx context.Context, name string) (docker.Image,
 }
 
 func (svc *Service) BuildTasks(ctx context.Context, repos []*graphql.Repository, spec *batches.BatchSpec) ([]*executor.Task, error) {
-	workspaceConfigs := []batches.WorkspaceConfiguration{}
-	for _, conf := range spec.Workspaces {
-		g, err := glob.Compile(conf.In)
-		if err != nil {
-			return nil, err
-		}
-		conf.SetGlob(g)
-		workspaceConfigs = append(workspaceConfigs, conf)
+	builder, err := executor.NewTaskBuilder(spec, svc)
+	if err != nil {
+		return nil, err
+	}
+
+	return builder.BuildAll(ctx, repos)
+}
+
+type repoWorkspaces struct {
+	paths              []string
+	onlyFetchWorkspace bool
+}
+
+// findWorkspaces matches the given repos to the workspace configs and
+// searches, via the Sourcegraph instance, the locations of the workspaces in
+// each repository.
+// The repositories that were matched by a workspace config are returned in
+// workspaces. root contains the repositories that didn't match a config.
+// If the user didn't specify any workspaces, the repositories are returned as
+// root repositories.
+func (svc *Service) findWorkspaces(
```

Comment:
Is this used anywhere now?

Reply:
Nice catch! More - lines for the diff gods!